### PR TITLE
Removing crypto code and just relying on oauth2client helpers.

### LINKED ISSUE #1408: Move bytes signing code upstream into oauth2client
Code was last modified:
https://github.com/GoogleCloudPlatform/gcloud-python/pull/1338

Both `PyCrypto` and `pyOpenSSL` versions could be supported there as well as pure Python.

### PR DESCRIPTION
**Pull request by dhermes**
Fixes #1408.

-----

Woohoo it feels so good to delete so much code!

This PR is "just for display" until the next release of `oauth2client`, since it relies on https://github.com/google/oauth2client/pull/421.

/cc @nathanielmanistaatgoogle

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -9,6 +9,7 @@ deps =
     nose
     nose-exclude
     unittest2
+    -egit+https://github.com/google/oauth2client.git#egg=oauth2client
```
Comment (dhermes) [inline tox.ini]: This will need to be dropped before this PR can be merged.

Comment (tseaver) [inline tox.ini]: We're going to need to pin `oauth2client>=2.1.0` (or whatever version gets released with this feature) somewhere, no?

Otherwise, WNTL?

Comment (dhermes) [inline tox.ini]: Sorry I am a n00b, what is WNTL?

And yes, we'll need a pin once it gets released.

Comment (tseaver) [inline tox.ini]: "What's not to like"?

Comment (dhermes) [inline tox.ini]: Ha sorry. I'm fine just waiting to merge this until an `oauth2client` release gets cut. (We can push to get one soon.)

**Comment by dhermes** [comment]
Waiting for [`0.2.1` tag][1] to build, then will update this PR with `oauth2client>=0.2.1` and we can get it merged.

[1]: https://travis-ci.org/google/oauth2client/builds/113466995

**Comment by dhermes** [comment]
OK https://pypi.python.org/pypi/oauth2client/2.0.1 is out.

Updated this PR. @tseaver PTAL.

**Comment by tseaver** [comment]
Ugh, they pushed out a new feature in a `2.0.1` release?

Anyhow, LGTM.

**Comment by dhermes** [comment]
They was me. It's not a new feature. `ServiceAccountCredentials.sign_blob` already existed, it just wasn't implemented for all possible credential types.

**Pull request merged**